### PR TITLE
tests: enable debian sid as part of the main suite on travis

### PR DESCRIPTION
**Pull request by sergiocazzolato**

### CODE REVIEW

**Comment by bboozzoo** [comment]
Looks like we'd need to disable the unit tests on sid:
```
2019-01-21 20:42:07 Error executing google:debian-sid-64:tests/unit/go : 
-----
+ mkdir -p /tmp/static-unit-tests/src/github.com/snapcore
+ cp -ar /home/gopath/src/github.com/snapcore/snapd /tmp/static-unit-tests/src/github.com/snapcore
+ chown -R test:12345 /tmp/static-unit-tests
+ rm -r /tmp/static-unit-tests/src/github.com/snapcore/snapd/vendor/github.com/ /tmp/static-unit-tests/src/github.com/snapcore/snapd/vendor/golang.org/ /tmp/static-unit-tests/src/github.com/snapcore/snapd/vendor/gopkg.in/
+ rm -rf /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/autom4te.cache /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/configure /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/test-driver /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/config.status /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/config.guess /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/config.sub /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/config.h.in /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/compile /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/install-sh /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/depcomp /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/build /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/missing /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/aclocal.m4 /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/Makefile /tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/Makefile.in
+ su -l -c 'cd /tmp/static-unit-tests/src/github.com/snapcore/snapd && PATH=/home/gopath/bin:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin:/usr/lib/go-1.6/bin:/var/lib/snapd/snap/bin GOPATH=/tmp/static-unit-tests ./run-checks --static' test
Obtaining dependencies
Checking docs
Checking formatting
Formatting wrong in following files:
/tmp/static-unit-tests/src/github.com/snapcore/snapd/snap/gadget_test.go
/tmp/static-unit-tests/src/github.com/snapcore/snapd/progress/ansimeter_test.go
/tmp/static-unit-tests/src/github.com/snapcore/snapd/osutil/udev/netlink/uevent_test.go
/tmp/static-unit-tests/src/github.com/snapcore/snapd/osutil/udev/netlink/uevent_test.go
/tmp/static-unit-tests/src/github.com/snapcore/snapd/interfaces/policy/basedeclaration_test.go
/tmp/static-unit-tests/src/github.com/snapcore/snapd/interfaces/policy/basedeclaration_test.go
/tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/snap/cmd_run_test.go
/tmp/static-unit-tests/src/github.com/snapcore/snapd/cmd/snap/cmd_run_test.go
/tmp/static-unit-tests/src/github.com/snapcore/snapd/asserts/account_key_test.go
```

File: tests/lib/journalctl.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ get_last_journalctl_cursor(){
 }
 
 start_new_journalctl_log(){
-    echo "New test starts here - $SPREAD_JOB" | systemd-cat -t snapd-test
+    echo "New test starts here - $SPREAD_JOB" | systemd-cat -t snapd-test -p alert
```
Comment (mvo5) [inline tests/lib/journalctl.sh]: AFAICT we don't need the changes in this file - this is all related to systemd/systemd#11502 it seems

File: tests/lib/prepare-restore.sh [hunk 2]
```diff
@@ -416,6 +416,11 @@ prepare_suite_each() {
     # back test directory to be restored during the restore
     tar cf "${PWD}.tar" "$PWD"
 
+    # WORKAROUND for memleak https://github.com/systemd/systemd/issues/11502
+    if [[ "$SPREAD_SYSTEM" == debian-sid* ]]; then
+        systemctl restart systemd-journald
```
Comment (zyga) [inline tests/lib/prepare-restore.sh]: Fun :)

**Comment by mvo5** [comment]
We may need to remove etckeeper from the image, I did that in https://github.com/snapcore/snapd/pull/6413 and it helped to make things more reliable. Having said this, we probably also need to find out why etckeeper seems to badly interact with the apt hook we install from snapd.

**Comment by mvo5** [comment]
With 6413 being mergd we can close this one now.

**Pull request closed without merging**